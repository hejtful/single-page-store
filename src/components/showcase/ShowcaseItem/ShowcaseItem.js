import React from 'react';

import { locals } from 'locals';

import styles from './ShowcaseItem.module.css';

export const ShowcaseItem = ({
  id,
  title,
  price,
  image,
  description,
  onAdd,
}) => {
  function handleAdd() {
    onAdd(id);
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.imageWrapper}>
        <img className={styles.image} src={image} alt={`${title} product`} />
      </div>

      <div className={styles.infoWrapper}>
        <div className={styles.titleWrapper}>
          <h4 className={styles.title}>{title}</h4>
          <span>{price}$</span>
        </div>

        <p className={styles.description}>{description}</p>

        <button type="button" onClick={handleAdd}>
          {locals.showcase.add_to_order}
        </button>
      </div>
    </div>
  );
};
