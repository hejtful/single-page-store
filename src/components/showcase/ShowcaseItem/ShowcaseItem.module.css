.wrapper {
  border: 1px solid #dbdbdb;
  display: flex;
  margin-top: 1em;
  padding: 0.5em;
}

.imageWrapper {
  flex: 0 0 30%;
  margin-right: 0.5em;
}

.image {
  max-width: 100%;
}

.infoWrapper {
  flex-grow: 1;
}

.titleWrapper {
  align-items: flex-start;
  display: flex;
  justify-content: space-between;
}

.title {
  margin: 0;
}

.description {
  font-size: 0.875em;
  margin: 0.5em 0;
}
