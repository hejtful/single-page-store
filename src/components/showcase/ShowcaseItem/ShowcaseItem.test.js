import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ShowcaseItem } from './ShowcaseItem';

const props = {
  id: '1',
  title: 'Test title',
  price: '1.25',
  image: 'http://test.image.com',
  description: 'A bit shorter description',
};

const methods = {
  onAdd: jest.fn(),
};

it('renders with props and add button', () => {
  const { container } = render(<ShowcaseItem {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('should call onAdd method with item ID when add button is clicked', () => {
  const { getByRole } = render(<ShowcaseItem {...props} {...methods} />);

  userEvent.click(getByRole('button'));

  expect(methods.onAdd).toBeCalledWith(props.id);
});
