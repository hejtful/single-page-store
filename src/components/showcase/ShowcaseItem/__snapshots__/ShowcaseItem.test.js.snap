// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props and add button 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="Test title product"
        class="image"
        src="http://test.image.com"
      />
    </div>
    <div
      class="infoWrapper"
    >
      <div
        class="titleWrapper"
      >
        <h4
          class="title"
        >
          Test title
        </h4>
        <span>
          1.25
          $
        </span>
      </div>
      <p
        class="description"
      >
        A bit shorter description
      </p>
      <button
        type="button"
      >
        Add to Order
      </button>
    </div>
  </div>
</div>
`;
